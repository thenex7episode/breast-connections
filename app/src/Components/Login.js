import React, { Component } from 'react';

export default class Login extends Component {
    constructor() {
        super()

        this.state = {
            user: null,
            message: null
        }
    }


    login = e => {
        console.log(e.key)
        this.setState({message: null})
        const username = this.refs.username.value
        const password = this.refs.password.value
        axios.post('/login', {
            username,
            password
        }).then(response => {
            this.setState({user: response.data})
            this.props.histroy.push('/profile')
        })
    }

    render() {
        return (
            <div>
                <input type="text"/>
                <input type="password"/>
            </div>
        );
    }
}