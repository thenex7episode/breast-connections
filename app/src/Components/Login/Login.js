import React, { Component } from 'react';
import axios from 'axios'

export default class Login extends Component {
    constructor() {
        super()

        this.state = {
            user: null,
            message: null
        }
        this.login = this.login.bind(this)
    }


    login= (e) => {
        console.log(e.key)
        this.setState({message: null})
        const username = this.refs.username.value;
        const password = this.refs.password.value
        console.log('-------username',password)
        axios.post('/login', {
            username,
            password
        }).then(response => {
            console.log('---------R.DATA',response.data)
            this.setState({ user: response.data})
            this.props.history.push('/profile')
            this.props.isLoggedIn()
           
        }).catch(error => {
            this.setState({ message: 'Something went wrong'})
        })
    }
    loginOnEnter(e) {
        if(e.key === 'Enter') {
            console.log(e.key)
            this.setState({message: null})
            const username = this.refs.username.value
            const password = this.refs.password.value
            axios.post('/login', {
                username,
                password
            }).then(response => {
                console.log('-----r.data',response.data)
                this.setState({user: response.data})
                this.props.history.push('/profile')
                this.props.isLoggedIn()
            }).catch(error => {
                this.setState({message: 'Something went wrong'})
            })
        }
    }
    
    render() {
        console.log('--------state', this.state)
        return (
            <div onKeyPress= {e => this.loginOnEnter(e)}>
                <h1>Login</h1>
                <input type="text" ref='username'/>
                <input type="password" ref='password'/>
                <button onClick = {e => this.login(e)}>Login</button>
            </div>
        );
    }
}