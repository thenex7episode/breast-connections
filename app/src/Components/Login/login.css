/* @media only screen and (max-width: 768px) {
    .d-login{
       margin-top:17em;

    }
} */