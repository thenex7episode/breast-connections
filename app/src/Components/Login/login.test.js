import Login from './Login'

it("renders correctly", () => {
    const wrapper = shallow(
        <Login />
    );
    expect(wrapper).toMatchSnapshot()
});