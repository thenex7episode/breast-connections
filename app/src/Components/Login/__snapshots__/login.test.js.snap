// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="d-login"
  onKeyPress={[Function]}
  style={
    Object {
      "padding": "5em",
    }
  }
>
  <h1>
    Login
  </h1>
  <Input
    disabled={false}
    onChange={[Function]}
    placeholder="username"
    prefixCls="ant-input"
    type="text"
  />
  <Input
    disabled={false}
    onChange={[Function]}
    placeholder="password"
    prefixCls="ant-input"
    type="password"
  />
  <Button
    ghost={false}
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
    type="primary"
  >
    Login
  </Button>
  <Link
    replace={false}
    to="/register"
  >
    <Button
      ghost={false}
      loading={false}
      prefixCls="ant-btn"
    >
      Register
    </Button>
  </Link>
</div>
`;
